File: samples/sample.py
#!/bin/env python
# -*- coding: utf-8 -*-
import re

# single line comment.
"""
Here is multiple line comment.
"""

class ClassA:
    def __init__(self, arg1, arg2=1, arg3=None):
        self.value = arg1 + arg2
        if (arg3 is not None):
            self.value += arg3

class ClassB(ClassA):
    def __init__(self):
        super(ClassB, self).__init__(1, 2, 3)

    def function(self):
        return 6


var = ClassB()
print(str(var.value))

if (not (var.value == var.function())):
    raise

if (re.search(r"^m([itlane]+e)$", "melanite")):
    print("matched")

try:
    x = 1 / 0
except ZeroDivisionError as e:
    print(e)

File: samples/sample.rb
#!/bin/env ruby
# -*- coding: utf-8 -*-
require "time"

# single line comment.
=begin
Here is multiple line comment.
=end

CONST_VARIABLE = 1

class ClassA
  def initialize(arg1, arg2=1, arg3=nil)
    @value = arg1 + arg2
    @value += arg3 unless (arg3.nil?)
  end
end

class ClassB < ClassA
  attr_reader :value

  def initialize()
    super(1, 2, 3)
  end

  def function()
    return 6
  end
end

$global_var = ClassB.new()
puts($global_var.value)
raise unless ($global_var.value == $global_var.function())

if ("melanite" =~ /^m([itlane]+e)$/)
  puts("matched")
end

begin
  x = 1 / 0
rescue ZeroDivisionError => e
  puts(e)
end

puts(`ls`)
